File: Week 6/Sentimental - readability/readability.py
from cs50 import get_string

def main():
    text = get_string("Text: ")

    letters = 0
    words = 0
    sentences = 0

    # Count letters, words, sentences
    for i, char in enumerate(text):
        if char.isalpha():
            letters += 1

        if char in [".", "!", "?"]:
            sentences += 1

        # Count words (space-based)
        if char == " ":
            words += 1

    # Last word
    words += 1

    # Coleman-Liau components
    L = (letters / words) * 100
    S = (sentences / words) * 100

    index = round(0.0588 * L - 0.296 * S - 15.8)

    # Output formatting
    if index < 1:
        print("Before Grade 1")
    elif index >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {index}")


if __name__ == "__main__":
    main()

File: Week 2/Readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main(void)
{
    // Prompt user for text
    string text = get_string("Text: ");

    int letters = 0;
    int words = 1;     // Start at 1 since the last word may not have a space
    int sentences = 0;

    // Count letters, words, sentences
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        char c = text[i];

        // Letters
        if (isalpha(c))
        {
            letters++;
        }
        // Words
        else if (c == ' ')
        {
            words++;
        }
        // Sentences
        else if (c == '.' || c == '!' || c == '?')
        {
            sentences++;
        }
    }

    // Calculate L and S
    float L = ((float)letters / words) * 100;
    float S = ((float)sentences / words) * 100;

    // Coleman-Liau index
    float index = 0.0588 * L - 0.296 * S - 15.8;
    int grade = round(index);

    // Print grade
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }

    return 0;
}
